using_restricted_array.rb

```diff
@@ -8,28 +8,94 @@
 
 ## Calculates the length of the restricted integer array_size
 def length(array)
-  puts "NOT IMPLEMENTED"
+  if array[0] == nil
+    return nil
+  else
+    count = 0
+    i = 0
+    while (array[i] != nil)
+        count += 1
+        i += 1
+    end
+  return count
+  end
 end
 
 # Prints each integer values in the array
 def print_array(array)
-  puts "NOT IMPLEMENTED"
+  if array[0] == nil
+    print "This is an empty array."
+  else
+    i = 0
+    while (array[i] != nil)
+      print "\n"
+      print array[i]
+      i += 1
+    end
+  end
 end
 
 # Reverses the values in the integer array
 def reverse(array, length) # Ruby
-  puts "NOT IMPLEMENTED"
+  if array[0] == nil
+    print "This is an empty array."
+  else
+    print "\n"
+    i = 0
+    j = length - 1
+    while i < j
+      temp  = array[i]
+      array[i] = array[j]
+      array[j] = temp
+      i += 1
+      j -= 1
+    end
+  end
+  return array
 end
 
 # For an unsorted array, searches for 'value_to_find'.
 # Returns true if found, false otherwise.
 def search(array, length, value_to_find)
-  puts "NOT IMPLEMENTED"
+  if array[0] == nil
+    print "This is an empty array."
+  else
+    i = 0
+    while (array[i] != nil)
+      if array[i] == value_to_find
+        return true
+      else
+        i += 1
+      end
+    end
+    return false
+  end
 end
 
 # Sorts the array in ascending order.
 def sort(array, length)
-  puts "NOT IMPLEMENTED"
+  if length == 0
+    print "This is an empty array."
+  else
+    i = 0
+    until i == length - 1
+      j = i + 1
+      min_index = i
+      until j == length
+        if array[min_index] > array[j]
+          min_index = j
+          j += 1

```

Comment:
Minor: you could end the conditional statement after updating `min_index`. And have  `j += 1` outside the conditional since it applies in both the if case and the else case.